src/flownet/network_model/_network_model.py

```diff
@@ -50,6 +50,8 @@ def __init__(
         if isinstance(fault_planes, pd.DataFrame):
             self._fault_planes: pd.DataFrame = fault_planes
             self._faults: Optional[Dict] = self._calculate_faults(fault_tolerance)
+        else:
+            self._faults = None
```

Comment:
Some minor thoughts: maybe, for completeness, use typing in this part of the if-else-statement. Could also a good idea to set _fault_planes to None for completeness. All not strictly necessary, but not doing this in the first place has led to the error you had to fix.

```suggestion
        else:
            self._fault_planes: Optional[pd.DataFrame] = None
            self._faults: Optional[Dict] = None            
```

Reply:
Yes I tried 
` self._faults: Optional[Dict] = None   `
and I work in my computer but I get a problem inside the checking test that the server was doing inside github for running flownet. Something about ` self._faults:` was declared as `Optional[Dict]` before. I'll add this changes to where I get the problem exactly.

Reply:
This is the message given by the server that is runing flownet          
```
Check code style & linting
25s
##[error]Process completed with exit code 1.
Run source $VENV_PATH/bin/activate
All done! ✨ 🍰 ✨
51 files would be left unchanged.

------------------------------------
Your code has been rated at 10.00/10

src/flownet/network_model/_network_model.py:54: error: Attribute '_fault_planes' already defined on line 51
src/flownet/network_model/_network_model.py:55: error: Attribute '_faults' already defined on line 52
src/flownet/network_model/_network_model.py:226: error: Value of type "Optional[Any]" is not indexable
src/flownet/network_model/_network_model.py:235: error: Item "None" of "Optional[Any]" has no attribute "loc"
src/flownet/network_model/_network_model.py:235: error: Value of type "Optional[Any]" is not indexable
Found 5 errors in 1 file (checked 50 source files)
##[error]Process completed with exit code 1.
```

Reply:
Maybe this should be the way of writing,

```python
self._fault_planes: Optional[pd.DataFrame] = None
self._faults: Optional[Dict] = None
if isinstance(fault_planes, pd.DataFrame):
            self._fault_planes = fault_planes
            self._faults = self._calculate_faults(fault_tolerance)
```
to solve 
```
src/flownet/network_model/_network_model.py:54: error: Attribute '_fault_planes' already defined on line 51
src/flownet/network_model/_network_model.py:55: error: Attribute '_faults' already defined on line 52
```
Then, to solve,
```
src/flownet/network_model/_network_model.py:226: error: Value of type "Optional[Any]" is not indexable
src/flownet/network_model/_network_model.py:235: error: Item "None" of "Optional[Any]" has no attribute "loc"
src/flownet/network_model/_network_model.py:235: error: Value of type "Optional[Any]" is not indexable
```
you'll have to rewrite

https://github.com/equinor/flownet/blob/0cf8c5ba89ff5136f486f5eb22780e733bf52251/src/flownet/network_model/_network_model.py#L226-L229

to something like:
```python
if self._fault_planes is not None:
    fault_names = self._fault_planes["NAME"].unique().tolist()
        if not fault_names:
            return None
```

**NB.** I didn't test this code; so it might not be completely accurate.